### PR TITLE
formatting markdown files and add doc for globalid, dal-php, dal-cpp, dal-globalid

### PR DESCRIPTION
**Pull request by enterprising**
1、markdown语法完善；  
2、更新globalID原理；  
3、更新php接入dal；  
4、更新c++接入dal；
5、更新dal globalid接入及相关说明。

### CODE REVIEW

File: _docs/DAL接入/sharding-limit.md [hunk 1]
```diff
@@ -28,126 +28,125 @@ permalink: /docs/sharding-limit/
    如果update更新了shardingkey的值，改变了data row与表的分片规则，比如原值是10，update成了11，按照路由规则就变了，但是这条记录还在10对应的分表里面，当业务按照新的值11去查询这条记录的时候，就会发现查询不到这条记录(因为update之后记录还在10对应的表里面) .
 
 9. 对于mapping方案，需要注意3点:
-   a.采用mapping方案的前提是mappingkey和shardingkey是一对一或者多对一的对应关系，不支持一对多或多对多的关系。
+   a.采用mapping方案的前提是mappingkey和shardingkey是一对一或者多对的对应关系，不支持一对多或多对多的关系。
```
Comment (hugowaitforit) [inline _docs/DAL接入/sharding-limit.md:31]: 应该是”多对一“吧

Comment (enterprising) [reply _docs/DAL接入/sharding-limit.md:31]: 。。对，晚上我再改改

**Comment by enterprising** [comment]
@hugowaitforit 静哥，这个pr我加了在之前的基础上点东西，有空看下  @kong0812qiao @blueswhisper @today-isaniceday

File: _docs/DAL原理/globalid.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## GloballD原理
+
+为满足不同业务场景的需求，Athena开发了多种不同种类的Globalld提供给业务使用。大部分返回的Globalld都是类型的结果，基本原理都是在用户的数据库中创建一张 `dal_sequences` 表，然后在该表中写入记录，存储种子值，每个Athena进程在有请求要获取Globalld时会从对应的dal_sequences表中获取对应记录的种子值作为填充位填充到Globalld的高位中保证Globalld的全局唯一性， 也可以通过控制种子值的范围来控制最终生成的Globalld的范围，在Globalld的低位写入不同的信息用于不同的目的。最后将拼凑好的Globalld返回给Client。
+
+同一个DAL Group可以有多个GloballD，多个Globalld的种子值写在同一个Home库中的dal_sequences表中。Home库对于非Sharding的库就等于业务自己的数据库，Sharding库即便Sharding后，依然有些SQL无法找到合适的库执行，而且业务大概率是Sharding表和非Sharding表混用的，原来未sharding的表会存在home库上。此处所说的库是database概念，不是数据库进程级别，所以Home库可以位于独立的数据库机器，同机器不同数据库进程，同进程不同database，甚至可以与某个Sharding分片库相同。业务在请求同一个Athena集群的不同节点时，这些节点都会去同一个dal sequences表获取最新的种子值，每次获取使用 `select for update` 语法保证每次获取种子只会有一个Athena节点成功来避免Globalld重复。
+
+出于性能的考虑，每个Athena节 点会缓存部分的种子值，只有种子值用尽后才会去dal sequences表中获取一段新的种子值缓存。
+
+为了支持多机房部署，同一个逻辑的Globalld在不同机房会使用不同的种子范围，这些范围是保证分段不重合的。也就是说，对于同一个Globalld的种 子在dal_sequences表中会有多条记录，不同的机房使用不同的种子范围。结合下面的种子样例，实际的种子名是逻辑种子名+ezone后缀作为seq name存入的。不同ezone使用不同的种子。
```
Comment (blueswhisper) [inline _docs/DAL原理/globalid.md:22]: ezone is internal term. for general, **rack** is better choose for comprehension.

File: _docs/DAL原理/mysql-handshake.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+title: MySQL连接协议
+permalink: /docs/mysql-handshake/
+---
+
+# 连接过程
+
+1. client通过tcp协议连接上server的监听端口建立TCP连接。
+2. server发送Handshake给client告知协议版本，分配给client的connectio_id, server的兼容性flag，给client加密密码的盐,加密算法
+3. client通过接收的Handshake包中的信息计算要传输给server的client端兼容性flag，登录用户名，使用server端给的盐加密密码的密文，要登录的database等信息拼装为HandshakeResponse包发送给server
+4. server接收到HandshakeResponse进行校验，认证成功给client回OK包，认证失败给client回ERR包
```
Comment (blueswhisper) [inline _docs/DAL原理/mysql-handshake.md:11]: Link to upstream docs is the better choose. Upstream may has some upgrade on protocol.

**Pull request merged**